mne/io/base.py

```diff
@@ -2503,12 +2503,14 @@ def _check_raw_compatibility(raw):
     for ri in range(1, len(raw)):
         if not isinstance(raw[ri], type(raw[0])):
             raise ValueError(f'raw[{ri}] type must match')
-        for key in ('nchan', 'bads', 'sfreq'):
+        for key in ('nchan', 'sfreq'):
             a, b = raw[ri].info[key], raw[0].info[key]
             if a != b:
                 raise ValueError(
                     f'raw[{ri}].info[{key}] must match:\n'
                     f'{repr(a)} != {repr(b)}')
+        if not set(raw[ri].info['bads']) == set(raw[0].info['bads']):
```

Comment:
I'm surprised no style checker complained about this, so I'll have to do it :)
```suggestion
        if set(raw[ri].info['bads']) != set(raw[0].info['bads']):
```
A little BUG entry in `doc/changes/latest.inc` would be good, too!